tekton/release-cheat-sheet.md

```diff
@@ -11,52 +11,70 @@ need a checkout of the operator repo, a terminal window and a text editor.
-       commit identified in step1.
-
-   ```bash
-   TEKTON_RELEASE_BRANCH=release-v0.62.x git checkout -b ${TEKTON_RELEASE_BRANCH}
-   ```
-    2. If this is a **patch release** make sure that the correct branch is checkout. eg: If we are making release
-       v0.62.1, then make sure the `release-v0.62.x` is checked out.
-
-5. Make sure the Tektoncd Component versions are the one you want, in
+3. Make sure the Tektoncd Component versions are the one you want, in
    `components.yaml`. Those are kept up-to-date by our bots, but just
    in case.
 
-6. minor version release vs patch release:
-    1. if this is a minor version release push the branch to `github.com/tektoncd/operator`
 
-    ```bash
-    git push tektoncd ${TEKTON_RELEASE_BRANCH}
-    ```
+4. Set the release branch and update yaml files
+   
+   1. **For Minor Version Release:** Use the script to create and set up the release branch. Example
+      ```bash
+         TEKTON_RELEASE_BRANCH=release-v0.62.x
+         # ./hack/release-setup-branch.sh <old release version> <new release version>
+         ./hack/release-setup-branch.sh devel ${TEKTON_RELEASE_VERSION}
+      ```
+      The script will automatically create a new branch for the minor version (e.g., release-v0.62.x) and switch to it.
+      It updates the yaml files with the TEKTON_RELEASE_VERSION
+   
+   1. **For Patch Version Release:** Use the script to set up the release branch. Example
+      ```bash
+         TEKTON_RELEASE_BRANCH=release-v0.62.x
+         # ./hack/release-setup-branch.sh <old release patch version> <new release patch version>
+         ./hack/release-setup-branch.sh 0.62.0 0.62.1
+      ```
+      The script will automatically switch to the branch version. It updates the yaml files with the new release patch version.
+      It convey the user to create a PR with patch changes to the release branch
+
+5. **For Patch Version Release:** Ensure that the pull request created earlier has been merged before proceeding.
+
+6. Update Helm charts with the new version. If applicable, update CRDs in the charts with new types.  
+   ```bash
+   # Update labels in YAML files under templates directory
+   find charts/tekton-operator/templates -type f -name '*.yaml' -exec sed -i "s/operator\.tekton\.dev\/release: \"devel\"/operator.tekton.dev\/release: ${TEKTON_RELEASE_VERSION}/g" {} +
+   find charts/tekton-operator/templates -type f -name '*.yaml' -exec sed -i "s/version: \"devel\"/version: ${TEKTON_RELEASE_VERSION}/g" {} +
 
-    2. if this is a patch release, make a pull request to the appropriate minor version release branch (eg:
-       release-v0.62.x)
-       and get it merged before continuing to the next section.
+   # Update Chart.yaml
+   sed -i "s/^version: \"devel\"/version: ${TEKTON_RELEASE_VERSION}/" charts/tekton-operator/Chart.yaml
+   sed -i "s/^appVersion: \"devel\"/appVersion: ${TEKTON_RELEASE_VERSION}/" charts/tekton-operator/Chart.yaml
+  ```
```

Comment:
I think the charts should be updated first because the `release-setup-branch` will push a tag as well and we might have this workflow failed again https://github.com/tektoncd/operator/actions/runs/13387502750/job/37387396716

wdyt ?

Reply:
we need the branch before updating the chart, I added new step(7) after this to commit and create a PR for chart updates

Reply:
Yep, got it thanks 👍🏻